tests/components/motionmount/test_config_flow.py

```diff
@@ -399,28 +373,216 @@ async def test_show_zeroconf_form_new_ce_new_pro(
         data=discovery_info,
     )
 
-    assert result["step_id"] == "zeroconf_confirm"
     assert result["type"] is FlowResultType.FORM
-    assert result["description_placeholders"] == {CONF_NAME: "My MotionMount"}
+    assert result["step_id"] == "auth"
 
 
-async def test_zeroconf_device_exists_abort(
+async def test_authentication_incorrect_pin(
     hass: HomeAssistant,
     mock_config_entry: MockConfigEntry,
     mock_motionmount_config_flow: MagicMock,
 ) -> None:
-    """Test we abort zeroconf flow if device already configured."""
+    """Test that authentication is requested when needed."""
+    type(mock_motionmount_config_flow).name = PropertyMock(return_value=ZEROCONF_NAME)
+    type(mock_motionmount_config_flow).mac = PropertyMock(return_value=MAC)
+    type(mock_motionmount_config_flow).is_authenticated = PropertyMock(
+        return_value=False
+    )
+
+    user_input = MOCK_USER_INPUT.copy()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data=user_input,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "auth"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input=MOCK_PIN_INPUT.copy(),
+    )
+
+    assert result["type"] is FlowResultType.FORM
```

Comment:
All config flow tests should end in either FORM or CREATE_ENTRY to also test they are able to recover from an error

Reply:
I'm not sure I understand you.
This one does end in FORM, right?

Reply:
Sorry, I mean ABORT or CREATE_ENTRY

Reply:
But don't we want to test sub-parts of the flow? I think it's very powerful to test subparts in, well, parts. This makes troubleshooting a lot easier.
There are complete flows at the end of the test to make sure all things work together